pkgs/games/classicube/default.nix

```diff
@@ -75,6 +76,21 @@ stdenv.mkDerivation rec {
     wrapProgram "$out/bin/ClassiCube" \
       --run 'mkdir -p "$HOME/.local/share/ClassiCube"' \
       --run 'cd       "$HOME/.local/share/ClassiCube"'
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    mkdir -p "$out/share/applications" "$out/share/resources"
+    cp 'misc/CCicon.png' "$out/share/resources"
+    cat >"$out/share/applications/ClassiCube.desktop" <<EOF
+    [Desktop Entry]
+    Type = Application
+    Exec = "$out/bin/ClassiCube"
+    Icon = "$out/share/resources/CCicon.png"
+    Name = ClassiCube
+    GenericName = Sandbox Block Game
+    Categories = ["Game"]
+    EOF
```

Comment:
Instead of writing the desktop file manually, you could use `makeDesktopItem` and `copyDesktopItems` instead (I dont know where to find them in the docs, so ... [source](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/make-desktopitem/default.nix) and [example](https://github.com/NixOS/nixpkgs/blob/350fd0044447ae8712392c6b212a18bdf2433e71/pkgs/applications/graphics/ipe/default.nix))

I used the following patch and installed it via home-manager and tested on kde


<details>

```patch
--- orig.nix	2022-12-24 07:11:16.692263376 -0400
+++ default.nix	2022-12-24 07:05:38.361080499 -0400
@@ -3,6 +3,8 @@
 , fetchFromGitHub
 , dos2unix
 , makeWrapper
+, makeDesktopItem
+, copyDesktopItems
 , SDL2
 , libGL
 , curl
@@ -21,7 +23,19 @@
     sha256 = "sha256-m7pg9OL2RuCVKgFD3hMtIeY0XdJ1YviXBFVJH8/T5gI=";
   };
 
-  nativeBuildInputs = [ dos2unix makeWrapper ];
+  nativeBuildInputs = [ dos2unix makeWrapper copyDesktopItems ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      desktopName = pname;
+      genericName = "Sandbox Block Game";
+      exec = "ClassiCube";
+      icon = "CCicon";
+      comment = "Minecraft Classic inspired sandbox game";
+      categories = [ "Game" ];
+    })
+  ];
 
   prePatch = ''
     # The ClassiCube sources have DOS-style newlines
@@ -64,8 +78,14 @@
   postBuild = "cd -";
 
   installPhase = ''
+    runHook preInstall
+
     mkdir -p "$out/bin"
     cp 'src/ClassiCube' "$out/bin"
+
+    mkdir -p "$out/share/icons/hicolor/256x256/apps"
+    cp misc/CCicon.png "$out/share/icons/hicolor/256x256/apps"
+
     # ClassiCube puts downloaded resources
     # next to the location of the executable by default.
     # This doesn't work with Nix
@@ -75,6 +95,8 @@
     wrapProgram "$out/bin/ClassiCube" \
       --run 'mkdir -p "$HOME/.local/share/ClassiCube"' \
       --run 'cd       "$HOME/.local/share/ClassiCube"'
+
+    runHook postInstall
   '';
 
   meta = with lib; {

```

</details>

Reply:
Yeah, please do use that